ANMN/temperature_grid_product/anmn_temp_gridded_product.py

```diff
@@ -0,0 +1,565 @@
+    tmp_netcdf_dir          = tempfile.mkdtemp()
+    output_netcdf_file_path = os.path.join(tmp_netcdf_dir, generate_fv02_filename(df, nc_file_list))
+
+    input_netcdf_obj               = Dataset(nc_file_list[0], 'r')
+    output_netcdf_obj              = Dataset(output_netcdf_file_path, "w", format="NETCDF4")
+    output_netcdf_obj.date_created = datetime.now().strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    # read gatts from input, add them to output. Some gatts will be overwritten
+    input_gatts     = input_netcdf_obj.__dict__.keys()
+    gatt_to_dispose = ['author', 'toolbox_input_file', 'file_version', 'file_version_quality_control', 'quality_control_set',
+                       'CoordSysBuilder_', 'netcdf_filename', 'metadata', 'instrument_serial_number',
+                       'instrument_nominal_depth', 'compliance_checker_version', 'compliance_checker_last_updated',
+                       'geospatial_vertical_min', 'geospatial_vertical_max', 'featureType']
+
+    for gatt in input_gatts:
+        if gatt not in gatt_to_dispose:
+            setattr(output_netcdf_obj, gatt, getattr(input_netcdf_obj, gatt))
+
+    comment = 'comment: The following files have been used to generate the gridded product:\n%s' % " \n".join([os.path.basename(x) for x in nc_file_list])
+    setattr(output_netcdf_obj, 'comment', comment)
+    setattr(output_netcdf_obj, 'temporal_resolution', get_frequency_step_in_deployment(nc_file_list)[0] )
+    setattr(output_netcdf_obj, 'vertical_resolution', 1 )
+
+    instrument_nominal_depth = ", ".join(map(str, list_instrument_nominal_depth(nc_file_list)))
+    setattr(output_netcdf_obj, 'instrument_nominal_depth', instrument_nominal_depth)
+
+    output_netcdf_obj.createDimension("TIME", len(df.columns))
+    output_netcdf_obj.createDimension("DEPTH", len(df.index))
+    output_netcdf_obj.createDimension("LATITUDE", 1)
+    output_netcdf_obj.createDimension("LONGITUDE", 1)
+
+    var_time     = output_netcdf_obj.createVariable("TIME", "d", "TIME", fill_value=get_imos_parameter_info('TIME', '_FillValue'))
+    var_lat      = output_netcdf_obj.createVariable("LATITUDE", "d", "LATITUDE", fill_value=get_imos_parameter_info('LATITUDE', '_FillValue'))
+    var_lon      = output_netcdf_obj.createVariable("LONGITUDE", "d", "LONGITUDE", fill_value=get_imos_parameter_info('LONGITUDE', '_FillValue'))
+    var_depth    = output_netcdf_obj.createVariable("DEPTH", "f", "DEPTH", fill_value=get_imos_parameter_info('DEPTH', '_FillValue'))
+    var_lat[:]   = input_netcdf_obj['LATITUDE'][:]
+    var_lon[:]   = input_netcdf_obj['LONGITUDE'][:]
+    var_depth[:] = df.index.values
+
+    main_var           = 'TEMP'
+    fillvalue          = get_imos_parameter_info(main_var, '_FillValue')
+    output_main_var    = output_netcdf_obj.createVariable(main_var, "f4", ("TIME", "DEPTH"), fill_value=fillvalue)
+    output_main_var[:] = df.values.transpose()
+    output_main_var.coordinates = "TIME DEPTH"
```

Comment:
although the variable has 2 dimension here TIME DEPTH , the coordinates are TIME LONGITUDE LATITUDE DEPTH 
Please update the temperature unit to "degrees_Celsius"